### ISSUE TITLE
Add support for ENS domains

### ISSUE THREAD

**Issue by Zaczero** [LedgerHQ/ledger-live-desktop#2858]
- [x] I have checked this feature was not yet requested.

Currently when sending a transaction no ENS domain is being resolved. Instead I'm given the error *"not a valid ... address"*. Would love to see this dApp being implemented into Ledger Live! :-)

![200416041413_Ledger_Live_bc0c37](https://user-images.githubusercontent.com/10835147/79407021-c0863980-7f98-11ea-9324-936a3109bc02.png)

---

#### What is ENS?
https://ens.domains/

#### Why?
* Join the list of ENS-compatible wallets
* Stay ahead of the competing hardware wallets
* Simplify and speed up the send transaction process

#### Compatible coins/tokens *(multi-coin)*
* Ethereum
* Ethereum Classic
* Bitcoin
* Bitcoin Cash
* Litecoin
* XRP
* Dogecoin
* Monacoin

*... as of 16 Apr 2020*

#### Possible use of text records
There is this thing called text records which contains few standardized fields:
* Email
* URL
* Avatar
* Description
* Notice
* Keywords
* vnd.twitter - aka. Twitter username
* vnd.github - aka. GitHub username

I can only imagine how cool would it be to display those details while making a transaction inside the UI. A large circled avatar preview at the top, small email address below, some links (GitHub, Twitter) and a short description *(maybe?)*. **The potential is here.**

![200416042804_firefox_6ee793](https://user-images.githubusercontent.com/10835147/79408047-af3e2c80-7f9a-11ea-8895-fad8832878ba.png)

**Comment by Zaczero** [comment]
*Resolving ENS docs:*
https://docs.ens.domains/dapp-developer-guide/resolving-names

**Comment by joshgaweda** [comment]
This would be great!
ENS is critical for mass adoption!

**Comment by cyanlink** [comment]
yeah, unable to send to ENS domain from ledger live is really astonishing in 2020.

**Comment by gre** [comment]
How would you suggest the device to securely verify the address correctly correspond to that ENS domain? It's not as trivial as it seems.

**Comment by cyanlink** [comment]
> How would you suggest the device to securely verify the address correctly correspond to that ENS domain? It's not as trivial as it seems.

Okay first I assume you have the basic idea about ethereum and smart contracts. ENS system is a set of contracts (resolvers and registry and etc) that maintains the mapping record of (sub)domain and addresses and other info on the ethereum blockchain.

All Ledger Live Software has to do is to enable the ability to lookup ENS domains which is backed by ledger proxy API, show the user the corresponding address (the ledger client also need the ability to lookup for reverse record to display a human-readable ENS domain). The ledger nano device shows the address as usual so that user can check if there are MITM between ledger live and ledger nano. 

If you need to make sure that the lookup result is genuine and authentic, users have to trust infura/any other ethereum gateway provider such as ledger proxy.

**Comment by roddurd** [comment]
hello! I would very much still like this feature, and I'm sad that this conversation stopped and this issue is still open :( 

support for ENS will be great for adoption!!!

**Comment by Reuvens** [comment]
+1

**Comment by drakedanner** [comment]
Also interested in support for ENS.

**Comment by jamesmarkcook** [comment]
Came here to see if it was supported. +1

**Comment by 99Thebigdady** [comment]
i absolutely want that

**Comment by marcocattaneogotam** [comment]
+1

**Comment by max-arias** [comment]
+1

**Comment by Jb00gy** [comment]
+1

**Comment by a186r** [comment]
+1

**Comment by Oakoakoakoak** [comment]
+1

**Comment by willianpaixao** [comment]
+1

**Comment by Danielhighfive** [comment]
ENS support in 2022 is absolutely necessary! Its critical for mass adoption.

**Comment by pet125** [comment]
i really want that
